mavros_extras/src/plugins/companion_process_status.cpp

```diff
@@ -0,0 +1,84 @@
+/**
+ * @brief Companion Status plugin
+ * @file companion_status.cpp
+ * @author Tanja Baumann <tanja@auterion.com>
+ *
+ * @addtogroup plugin
+ * @{
+ */
+/*
+ * Copyright 2018 Tanja Baumann.
+ *
+ * This file is part of the mavros package and subject to the license terms
+ * in the top-level LICENSE file of the mavros repository.
+ * https://github.com/mavlink/mavros/tree/master/LICENSE.mdA
+ */
+
+#include <mavros/mavros_plugin.h>
+
+#include <mavros_msgs/CompanionProcessStatus.h>
+
+namespace mavros {
+namespace extra_plugins {
+
+//! Mavlink enumerations
+using mavlink::common::MAV_TYPE;
+using mavlink::common::MAV_STATE;
+using mavlink::common::MAV_COMPONENT;
+using utils::enum_value;
+
+/**
+ * @brief Obstacle companion process status plugin
+ *
+ * Publishes the status of components running on the companion computer
+ * @see status_cb()
+ */
+class CompanionProcessStatusPlugin : public plugin::PluginBase {
+public:
+	CompanionProcessStatusPlugin() : PluginBase(),
+	status_nh("~companion_process")
+	{ }
+
+	void initialize(UAS &uas_)
+	{
+		PluginBase::initialize(uas_);
+
+		status_sub = status_nh.subscribe("status", 10, &CompanionProcessStatusPlugin::status_cb, this);
+	}
+
+	Subscriptions get_subscriptions()
+	{
+		return { /* Rx disabled */ };
+	}
+
+private:
+	ros::NodeHandle status_nh;
+	ros::Subscriber status_sub;
+
+	/**
+	 * @brief Send companion process status to FCU over a heartbeat message
+	 *
+	 * Message specification: http://mavlink.org/messages/common#HEARTBEAT
```

Comment:
If this existed, the URL would be https://mavlink.io/en/messages/common.html#COMPANION_PROCESS_STATUS (ie not mavlink.org). But it does not exist.

Reply:
Hmm I had that first like that when we used our custom message. But when we decided to go for using the HEARTBEAT mavlink message instead of adding a custom one I thought it would be better to put that there instead. Is there a problem with that, should we change it?

Reply:
@baumanta the problem is not the msg name (which is ok). The problem is the link: it should be https://mavlink.io/en/messages/common.html#HEARTBEAT

Reply:
Yes, sorry, copy paste error. @TSC21 correct - mavlink.org is now defunct. Redirects were supposed to happen for messages but never did :-(

Reply:
@hamishwillee yeah this needs to be fixed, not only here, but in all other plugins with references to the message docs.

Reply:
Yes it sux. Probably not your highest priority to do globally, but worth fixing as we come to them.

Reply:
@hamishwillee @TSC21 Do I understand this right, we would just need to change all links to direct to mavlink.io instead of mavlink.org? But otherwise they are correct, no?

Reply:
Yes correct

Reply:
> But otherwise they are correct, no?

Yes, but note it isn't just a matter of changing the site address from .org to .io - there are other differences in the URI.